File: bash/fifo/tester.cs
using System;

public class Tester
{
    public static void Main(string[] args)
    {
        int numberToGuess = (new Random()).Next(1,101);
        Console.Error.WriteLine($"Tester  - Number to guess set to {numberToGuess}");
        int count = 0;
        int guess;
        string response;
        while (true) {
            response = Console.ReadLine();
            if (string.IsNullOrEmpty(response)) {
                continue;
            }
            Int32.TryParse(response, out guess);
            Console.Error.WriteLine($"Tester  - Handling input '{guess}'");
            count++;
            if (numberToGuess < guess) {
                Console.WriteLine("<");
            }
            else if (numberToGuess > guess) {
                Console.WriteLine(">");
            }
            else {
                Console.WriteLine("!");
                Console.Error.WriteLine($"Tester  - Number guessed in {count} attempts.");
                break;
            }
        }
    }
}

File: bash/fifo/tester.py
#!/usr/bin/env python3

import random
import sys

def printOnStderr(text):
    print(text, file=sys.stderr)

def run():
    numberToGuess = random.randint(1, 100)
    printOnStderr(f"Tester  - Number to guess set to {numberToGuess}")
    count = 0
    while True:
        guess = int(input())
        count+=1
        printOnStderr(f"Tester  - Handling input '{guess}'")
        if guess == numberToGuess:
            print("!")
            printOnStderr(f"Tester  - Number guessed in {count} attempts.")
            break
        if numberToGuess < guess:
            print("<")
        else:
            print(">")

if __name__=="__main__":
    run()
    
